### ISSUE TITLE
Unable to register security key at purelymail.com

### ISSUE THREAD

**Issue by FloatingMilkshake** [bitwarden/clients#7298]
### Steps To Reproduce

1. Go to purelymail.com
2. Create or log in to an account, then go to the Users tab and select any user
3. Under "Two-Factor Authentication (2FA)," click "Add Security Key"
4. Observe the error "Unable to register security key: TypeError. Please try again."

### Expected Result

Bitwarden's passkey dialog opens and prompts the user to create or use a passkey for Purelymail.

### Actual Result

An error is shown on the Purelymail page, and Bitwarden is not opened. Passkey creation fails.

### Screenshots or Videos

https://github.com/bitwarden/clients/assets/51096169/127c4999-743e-4cb4-851f-04c27faa39a6

### Additional Context

I first assumed issue https://github.com/bitwarden/clients/issues/6769 was related, but after updating to version 2023.12.0 I am able to create passkeys on porkbun.com, but not on purelymail.com. The full error in my console is:
```
Uncaught (in promise) TypeError: _b.authenticatorSelection is undefined
    create moz-extension://c9d9288a-ece3-434e-8198-3a4aff042ba6/content/fido2/page-script.js:16916
    page_script_awaiter moz-extension://c9d9288a-ece3-434e-8198-3a4aff042ba6/content/fido2/page-script.js:16858
    page_script_awaiter moz-extension://c9d9288a-ece3-434e-8198-3a4aff042ba6/content/fido2/page-script.js:16854
    create moz-extension://c9d9288a-ece3-434e-8198-3a4aff042ba6/content/fido2/page-script.js:16911
    create https://purelymail.com/shade.js line 1 > eval:244
    __async https://purelymail.com/shade.js line 1 > eval:37
    __async https://purelymail.com/shade.js line 1 > eval:21
    create https://purelymail.com/shade.js line 1 > eval:243
    <anonymous> https://purelymail.com/shade.js line 1 > eval:279
    <anonymous> https://purelymail.com/shade.js line 1 > eval:280
    <anonymous> https://purelymail.com/shade.js line 1 > eval:282
    makeEvalScope https://purelymail.com/shade.js:1
    evaluateScript https://purelymail.com/shade.js:1
    u https://purelymail.com/shade.js:1
page-script.js:16916:136
```

### Operating System

Windows, macOS

### Operating System Version

Windows 11 (Insider Dev build 23606.1000), macOS Sonoma 14.3

### Web Browser

Chrome, Safari, Microsoft Edge, Firefox

### Browser Version

Chrome version 120.0.6099.130; Safari version 17.3 (19617.2.1.11.3); Edge version 121.0.2277.4; Firefox version 121.0 (20231211174248)

### Build Version

2023.12.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by SergeantConfused** [comment]
Hi @FloatingMilkshake,

Thank you for this report.
1. Please deactivate the Passkey Storage function in the Bitwarden browser extension by navigating to Settings > Options > Ask to save and use passkeys, and then give this another try and let me know if the operating system's prompt to enrol a FIDO2 WebAuthn Security Key shows up.
2. If step 1 (above) does not change anything, please disable the extension entirely and give this another try and let me know how things go then.

This information would assist us in narrowing down the cause of this behaviour.

Thank you in advance,

**Comment by FloatingMilkshake** [comment]
Sorry, I thought I included a little more information in my original post! Disabling the Passkey Storage function does resolve the issue, as does disabling the extension entirely.

**Comment by SergeantConfused** [comment]
Hi @FloatingMilkshake,

Thank you. It sounds like it could be unrelated to Bitwarden. Are you able to enrol a FIDO2 WebAuthn Security Key as 2FA into that service via a different browser, maybe one which does not have Bitwarden installed as an extension inside it?


Thank you in advance,

**Comment by SergeantConfused** [comment]
Hi @FloatingMilkshake,

Thank you for your patience. This appears to be related to [this](https://github.com/bitwarden/clients/issues/6769) and it'll be addressed by this [PR](https://github.com/bitwarden/clients/pull/6799) which we aim to incorporate into a future release.

I will keep this GitHub report open for now.


All the best,

**Comment by FloatingMilkshake** [comment]
My apologies, I misunderstood and assumed that fix was already part of Bitwarden's 2023.12.0 release. Thanks!

**Comment by FloatingMilkshake** [comment]
Coming back to this to say the issue is resolved. Currently running Bitwarden version 2024.2.0, but it might have been resolved in a prior version; not sure. Either way, all good now!
